OpenRA.Mods.Common/FileFormats/WavReader.cs

```diff
@@ -206,5 +212,108 @@ protected override bool BufferData(Stream baseStream, Queue<byte> data)
 				return ++currentBlock >= numBlocks;
 			}
 		}
+
+		// Format docs https://wiki.multimedia.cx/index.php/Microsoft_ADPCM
+		public sealed class WavStreamMsAdpcm : ReadOnlyAdapterStream
+		{
+			static readonly int[] AdaptationTable = new[]
+			{
+				230, 230, 230, 230, 307, 409, 512, 614,
+				768, 614, 512, 409, 307, 230, 230, 230
+			};
+
+			static readonly int[] AdaptCoeff1 = new[] { 256, 512, 0, 192, 240, 460, 392 };
+
+			static readonly int[] AdaptCoeff2 = new[] { 0, -256, 0, 64, 0, -208, -232 };
+
+			readonly short channels;
+			readonly int blockDataSize;
+			readonly int numBlocks;
+
+			int currentBlock;
+
+			public WavStreamMsAdpcm(Stream stream, int dataSize, short blockAlign, short channels)
+				: base(stream)
+			{
+				this.channels = channels;
+				blockDataSize = blockAlign - channels * 7;
+				numBlocks = dataSize / blockAlign;
+			}
+
+			protected override bool BufferData(Stream baseStream, Queue<byte> data)
+			{
+				var bpred = new byte[channels];
+				var chan_idelta = new short[channels];
```

Comment:
Not sure about these variable names, not a big fan of using abbreviations. At least it should be named  `chanIdelta `